lib/DIContext.ts

```diff
@@ -94,3 +94,5 @@ export class DIContext<Dependencies extends object, ExternalDependencies = {}> {
     await this.awilixManager.executeInit()
   }
 }
+
+// Could we extract TestContext? don't have to be right now, just thinking about it
```

Comment:
will give it a shot, wonder how reusable/customizable it could be

Reply:
it was trickier than expected, but I think the end result is cleaner than what NST had